nmpolicy/internal/capture/capture.go

```diff
@@ -88,6 +92,12 @@ func (c Capture) Resolve(
 	return capturesState, nil
 }
 
+func (c Capture) ResolveCapturePath(
+	capturesCache map[string]types.CaptureState,
+	capturePath string) (interface{}, error) {
+	return nil, nil
+}
+
```

Comment:
The `Capturer` has finish his job since at the expander state we already have the `capturedState` we just need a function at the expander that will call lexer/parser/resolver to resolve the path from placeholder with the `capturedState`, you will have to use stubs for lexer/parser/resolver  and implement this function.

It should be something like:

```golang
func (e StateExpander) expandCaptureEntryPath(captureEntryPath string) (interface{}, error)
        captureEntryPathTokens, err := e.lexer.Lex(captureEntryPath)
        if err != nil { 
           return nil, err
        }
        captureEntryPathAST, err := e.parser.Parse(captureEntryPathTokens)
        if err != nil {
           return nil, err
        }
        
        expandedCaptureEntryPath, err := c.resolver.ResolveCaptureEntry(e.capturedState, captureEntryPathAST)      
       if err != nil {
           return nil, err
       }
       return expandedCaptureEntryPath, nil
}
```